providers/base/utils.go

```diff
@@ -0,0 +1,24 @@
+package base
+
+import (
+	"fmt"
+
+	providertypes "github.com/skip-mev/slinky/providers/types"
+)
+
+// createResponseCh creates the response channel for the provider.
+func (p *Provider[K, V]) createResponseCh() error {
```

Comment:
eventually, we can extend this to have a thread safe abstracted channel that we can verifiably know if it is closed or full before writes happen.